restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/ClientsEndpoint.java

```diff
@@ -296,8 +297,12 @@ private Client update(final Client client, final MultivaluedMap<String, String>
                 result = result.setVoiceApplicationSid(getSid("VoiceApplicationSid", data));
             }
         }
-        if (data.containsKey("IsPushEnabled") && getBoolean("IsPushEnabled", data)) {
-            result.setPushClientIdentity(client.getSid().toString());
+        if (data.containsKey("IsPushEnabled")) {
+            if (getBoolean("IsPushEnabled", data)) {
+                result = result.setPushClientIdentity(client.getSid().toString());
+            } else {
+                result = result.setPushClientIdentity(null);
```

Comment:
i guess this is the only meaningfull change of this PR. This method is strange, it receives a final Client not meant to be modified. Then we assigned to a new var without final, and then we returned the modified instance... not sure what is the reason for this, i know is not part of this PR. Wouldnt make more sense to declare argument as regular non final,rather than suggesting Client is not modified to be modified later...?